sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-full-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -1154,5 +1159,69 @@ describe(
         expect(isBypassActive).toBe(true);
       },
     );
+
+    it(
+      "Should create a notification for the ministry and student for a blocked disbursement when the total assessed award is 0" +
+        " and there are no previously existing notifications for the disbursement.",
+      async () => {
+        // Arrange
+        const { student, disbursementId } =
+          await createBlockedDisbursementTestData(db, {
+            offeringIntensity: OfferingIntensity.fullTime,
+            isValidSIN: true,
+            disbursementValues: [],
+          });
+        // Queued job.
+        const mockedJob = mockBullJob<void>();
+
+        // Act
+        const result = await processor.processECert(mockedJob.job);
+
+        // Assert
+        expect(result).toStrictEqual([
+          "Process finalized with success.",
+          "Generated file: none",
+          "Uploaded records: 0",
+        ]);
+        expect(
+          mockedJob.containLogMessages([
+            "Disbursement estimated awards do not contain any amount to be disbursed.",
+            `Creating notifications for disbursement id: ${disbursementId} for student and ministry.`,
+            `Completed creating notifications for disbursement id: ${disbursementId} for student and ministry.`,
+          ]),
+        ).toBe(true);
+        const notifications = await db.notification.find({
+          select: {
+            id: true,
+            user: { id: true },
+            notificationMessage: { id: true },
+          },
+          relations: { user: true, notificationMessage: true },
+          where: {
+            metadata: {
+              disbursementId,
+            },
+            dateSent: IsNull(),
```

Comment:
Is it not necessary to add to the `beforeEach` an update to set the `dateSent``for all `dateSent` null records to ensure the only `null` records are the ones produced in this E2E scenario?

Reply:
To ensure the integrity of the data produced in the E2E, `{metadata:{disbursementId}}` seems to be sufficient. IMO we can remove the dateSent in the criteria only. 

Let me know if I am missing something.

Reply:
I missed that. All good 😄